components/brave_today/browser/direct_feed_controller.cc

```diff
@@ -87,31 +87,32 @@ mojom::ArticlePtr RustFeedItemToArticle(const FeedItem& rust_feed_item) {
 using ParseFeedCallback = base::OnceCallback<void(absl::optional<FeedData>)>;
 void ParseFeedDataOffMainThread(const GURL& feed_url,
                                 const std::string& charset,
-                                std::string body_content,
+                                const std::string& body_content,
```

Comment:
Some combination of this not being `const &` and not making a copy when doing the `base::ConvertToUtf8AndNormalize` was causing this to break, but only when the url pointed to the site, not a feed. To be honest, I don't fully understand why - my first thought was that the break was being caused by converting the string to UTF8 in place, but that didn't have any effect.

I applied the changes from my UTF8 fix PR one by one and found out that it worked without the `const &` here. Interestingly, it was still broken in the `news.com` case, which was then fixed by not modifying the string in place.

This is quite surprising to me and I was wondering if someone with a bit more C++ experience could weigh in (maybe @sangwoo108?).

Reply:
Don't worry - I worked it out :laughing: Another function was trying to access the moved data. Fortunately the string was being cleared out rather than just being garbage.

Reply:
Yeah, use-after-use is difficult to find ;(